### ISSUE TITLE
Next branch dependency updates for all projects

### ISSUE THREAD

**Issue by endigo9740** [skeletonlabs/skeleton#3021]
We should go ahead and update all project dependencies and check for regressions.

This will include a notable update to Astro 5, which now utilizes Vite 6:
https://astro.build/blog/astro-5/

As well as migrating all projects to a minimum of Node 22, to satisfy and upcoming change on Vercel:

![Screenshot 2024-12-04 at 1 02 49 PM](https://github.com/user-attachments/assets/92b52392-f077-495c-843d-6bc239587414)

Once changes changes are pushed, all contributors should be encouraged to update to use Node 22 on their local machines.

**Comment by endigo9740** [comment]
@Hugos68 FYI, it looks like there was a notable change to `@astro/mdx` that is triggering two issues:

First, the `getCollections()` method in Astro is no longer returning a `slug` per item. If you hover `item` it still shows this as a required value, but this is missing when you actually log the value. This is affecting a number of components in our app, including:

- NavGrid.astro
- TableCookbook.astro
- Navigation.astro

We can substitute `item.slug` with `item.filePath` though as a fix.

![Screenshot 2024-12-04 at 1 51 29 PM](https://github.com/user-attachments/assets/c3cbb931-2a95-4bdb-b4f1-5dae28f0f272)

Second, when switching to *any* page that has MDX rendered, there is a very obvious flash of styled content. However if you switch to a page that does not include MDX (ex: /news) this does not occur. I'm guessing this is either related to some pre-rendering or some new issues between the layout & rendered MDX.

https://github.com/user-attachments/assets/b0eb2d60-6594-476e-9b55-3ab450461c19

I'd welcome your support in troubleshooting these if possible.

**Comment by endigo9740** [comment]
FYI I've pushed additional dependency updates and updated the Vercel apps for the v3 docs and v3 theme generator apps to utilized Node 22. I also pushed some additional dep updates to the React package that was missing prior (`pnpm outdated` didn't show these for some reason). I also resolved some new linting issues that popped up in the React package related to the use of the Context API feature in React - they seem to require these to be scaffolded outside the component files now.

Unfortunately I noticed we're also having issues with the current PR's CI checks...

The "check" step is showing this Astro-check error, even though Astro has yet to be updated. For some reason I cannot replicate this locally.

![Screenshot 2024-12-04 at 4 51 35 PM](https://github.com/user-attachments/assets/1db50c99-0896-4123-ac3b-a24209d3d72a)

Additionally the "lint" step is failing as well. Best I can tell it seems to be running `eslint v9` even though the monorepo root's package.json specifies `eslint v8`. The v9 updates has some major breaking changes, which do not seem trivial to implement so we're purposely trying to stick with v8 for now.

No idea why it's running v9, my local instance shows the same with `pnpm lint`, but the package.json clearly indicates the following:

```
"eslint": "^8.57.1",
```

![Screenshot 2024-12-04 at 4 52 44 PM](https://github.com/user-attachments/assets/0b18e1a1-2108-49e9-ac7b-31febb1faa08)

Similarly the "test" step is failing. It seems to indicate an issue with Zag within the Svelte component. Once again, I can't seem to replicate this when running `pnpm test` locally from the monorepo root.

![Screenshot 2024-12-04 at 4 54 57 PM](https://github.com/user-attachments/assets/8e412d46-ddde-4e77-859e-20406f7f6452)

I'm feeling frustrated with so many breaking issues, so I'm just going to pause here for the day. If anyone has any ideas on these I'm all ears.
